perfkitbenchmarker/benchmarks/mysql_service_benchmark.py

```diff
@@ -0,0 +1,773 @@
+               stdout, stderr)
+
+  results.append(sample.Sample(
+      SYSBENCH_RESULT_NAME_DATA_LOAD,
+      load_duration,
+      SECONDS_UNIT,
+      metadata))
+
+  # Now run the sysbench OLTP test and parse the results.
+
+  for i in xrange(1, 3):
+    # First step is to run the test long enough to cover the warmup period
+    # as requested by the caller. Then we do the "real" run, parse and report
+    # the results.
+    duration = 0
+    if i == 1 and FLAGS.sysbench_warmup_seconds > 0:
+      duration = FLAGS.sysbench_warmup_seconds
+      logging.info('Sysbench warm-up run, duration is %d', duration)
+    elif i == 2:
+      duration = FLAGS.sysbench_run_seconds
+      logging.info('Sysbench real run, duration is %d', duration)
+
+    if duration > 0:
+      run_cmd_tokens = ['sysbench',
+                        '--test=%s' % OLTP_SCRIPT_PATH,
+                        '--oltp_tables_count=%d' % FLAGS.oltp_tables_count,
+                        '--oltp-table-size=%d' % FLAGS.oltp_table_size,
+                        '--rand-init=%s' % RAND_INIT_ON,
+                        '--db-ps-mode=%s' % DISABLE,
+                        '--oltp-dist-type=%s' % UNIFORM,
+                        '--oltp-read-only=%s' % OFF,
+                        '--num-threads=%d' % FLAGS.sysbench_thread_count,
+                        '--percentile=%d' % FLAGS.sysbench_latency_percentile,
+                        '--report-interval=%d' %
+                        FLAGS.sysbench_report_interval,
+                        '--max-requests=0',
+                        '--max-time=%d' % duration,
+                        '--mysql-user=%s' % MYSQL_ROOT_USER,
+                        '--mysql-password="%s"' % MYSQL_ROOT_PASSWORD,
+                        '--mysql-host=%s' % vm.db_instance_address,
+                        'run']
+      run_cmd = ' '.join(run_cmd_tokens)
+      stdout, stderr = vm.RobustRemoteCommand(run_cmd)
+      logging.info('Sysbench results: \n stdout is:\n%s\nstderr is\n%s',
+                   stdout, stderr)
+
+    if i == 2:
+      # We only need to parse the results for the "real" run.
+      logging.info('\n Parsing Sysbench Results...\n')
+      _ParseSysbenchOutput(stdout, results, metadata)
+
+  return results
+
+
+def _RDSParseDBInstanceStatus(json_response):
+  """Parses a JSON response from an RDS DB status query command.
+
+  Args:
+    json_response: The response from the DB status query command in JSON.
+
+  Returns:
+    A list of sample.Sample objects.
+  """
+  status = ''
+  # Sometimes you look for 'DBInstance', some times you need to look for
+  # 'DBInstances' and then take the first element
+  if 'DBInstance' in json_response:
+    status = json_response['DBInstance']['DBInstanceStatus']
+  else:
+    if 'DBInstances' in json_response:
+      status = json_response['DBInstances'][0]['DBInstanceStatus']
+
+  return status
+
+
+class RDSMySQLBenchmark(object):
```

Comment:
I think you should think of a database as a resource just like a network or anything else.   Perhaps in a future refactor we'll separate this out.